### PR TITLE
Implement add remove

### PR DESCRIPTION
**Pull request by rotimiazeez**
# BOOKSTORE APP: Create and Remove Books🎉 

> In this step you will configure your Redux Store and write actions and reducer for adding and removing books. You will also add some UI elements to your components and make them dispatch actions. NOTE: editing an existing book and updating the progress is not part of this step. Neither is the styling of the application's components. 💪🏽 

Additional description of the project and its features. ➕ 🚀  

- Import createStore
- Import combineReducers - We will only write a reducer for 'books' in this step, but in the future, we might want to add 'categories' or other reducers as well.
- Import applyMiddleware - To use some handy Redux middleware, like redux logger. It's optional, but it will help you a lot in the development process. To install it, you'll need to run:
npm i --save redux-logger
- Import your booksReducer, which you are going to write in /src/redux/books directory.

## Built With ❤️ 

- HTML (HTML5 semantic tags, `ul`, `li`,  `button`)
- CSS
-  REACT & REDUX

## Deployed Preview 🚀 
[Live Preview](coming soon)

### NOTE TO REVIEWER 😉 ❤️  
 😃 👍🏽

### CODE REVIEW

File: src/components/books/addNewBook/AddNewBook.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { useDispatch } from 'react-redux';
+import { createBook } from '../../../redux/books/books';
+
+const AddNewBook = () => {
+  const dispatch = useDispatch();
+  const [bookInfo, setBookInfo] = useState({
+    item_id: '',
+    title: '',
+    category: '',
+  });
```
Comment (dasileker) [inline src/components/books/addNewBook/AddNewBook.js]: - [x] The `useState` is not showing anything after creating a new book, also the `completed`  logo is too big.


screencapture-localhost-3000-2021-09-30-12_09_33

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+// REDUCER
+const booksReducer = (state = initialState, action) => {
+  let newArr = [];
+  switch (action.type) {
+    case CREATE_BOOK:
+      newArr = [...state.books, action.payload];
+      return { ...state, books: newArr };
+    case REMOVE_BOOK:
+      newArr = state.books.filter((book) => book.id !== action.payload);
```
Comment (dasileker) [inline src/redux/books/books.js:24]: - [ ] The redux Store is not storing the books after adding them, when I refresh the page they vanish.

Capture web_30-9-2021_121631_localhost

File: src/components/books/Books.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  const { booksReducer } = useSelector((state) => state);
+  const { books } = booksReducer;
+  return (
+    <div className="books-page">
+      <ul className="book-list">
+        {books.map((book) => (
+          <li key={book.id} className="book-card">
+            <Book bookInfo={book} progress={Math.floor(Math.random() * 101)} />
+          </li>
+        ))}
```
Comment (dasileker) [inline src/components/books/Books.js:17]: - [x] Like I mentioned the `completed` circle is too big, please fix it.

File: src/components/books/addNewBook/AddNewBook.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  return (
+    <section className="add-new-book-section">
+      <h2>ADD NEW BOOK</h2>
+      <form onSubmit={(e) => onSubmit(e)}>
+        <input onChange={handleChange('title')} type="text" name="title" placeholder="Book title" value={bookInfo.title} />
+        <input onChange={handleChange('category')} type="text" name="category" placeholder="Author name" value={bookInfo.category} />
+        <button onClick={onClick} type="submit">ADD BOOK</button>
+      </form>
```
Comment (dasileker) [inline src/components/books/addNewBook/AddNewBook.js]: - [x] Your `handleChange`  in `category` but the place holder says `author`???, please fix it

File: src/components/books/addNewBook/AddNewBook.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  const handleChange = (name) => (e) => {
+    setBookInfo({ ...bookInfo, [name]: e.target.value });
+  };
+  const onClick = () => {
+    setBookInfo({ ...bookInfo, item_id: uuidv4() });
+  };
+  const onSubmit = (e) => {
+    e.preventDefault();
+    dispatch(createBook(bookInfo));
+    setBookInfo({ ...bookInfo, title: '', category: '' });
```
Comment (dasileker) [inline src/components/books/addNewBook/AddNewBook.js:23]: - [x] Like I mentioned the return is not showing the correct value, it's empty for the `title` but the categories are a mess. I see all of them  I should see only the selected one for my book.

**Review by dasileker — APPROVED** [review]
### **Hi @rotimiazeez,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **To Highlight**

- The PR is good and descriptive 👍.
- The Linters are passing 👍.
- The commit-msg is consistent 👍.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @rotimiazeez,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **To Highlight**

- The PR is good and descriptive 👍.
- The Linters are passing 👍.
- The commit-msg is consistent 👍.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/books/addNewBook/AddNewBook.js (hunk 1 above)
Comment (rotimiazeez) [reply src/components/books/addNewBook/AddNewBook.js]: Fixed

File: src/redux/books/books.js (hunk 2 above)
Comment (rotimiazeez) [reply src/redux/books/books.js:24]: I'm not working with the API yet I'm not sure this is a requirement for this milestone

File: src/components/books/Books.js (hunk 3 above)
Comment (rotimiazeez) [reply src/components/books/Books.js:17]: done 👍

File: src/components/books/addNewBook/AddNewBook.js (hunk 4 above)
Comment (rotimiazeez) [reply src/components/books/addNewBook/AddNewBook.js]: It's fixed now my bad 🙏

File: src/components/books/addNewBook/AddNewBook.js (hunk 5 above)
Comment (rotimiazeez) [reply src/components/books/addNewBook/AddNewBook.js:23]: It fixed now

File: src/redux/books/books.js (hunk 2 above)
Comment (dasileker) [reply src/redux/books/books.js:24]: It's not an api just the return of the value. But if you feel its not necessary just puah it for another review

**Comment by rotimiazeez** [comment]
Hi @dasileker 🙋‍♂️,
Thanks for the feedback 🙏. I made the necessary changes as pointed out, I hope I get approved now.

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         

## Thank you for implementing all the project requirements, you've done a great job :trophy:  . There is nothing else to say other than... it's time to merge it :champagne:  :tada: Congratulations

Cheers and Happy coding!clap computer wine_glass
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by rotimiazeez** [comment]
Hi @GzimAsani 👋,
I appreciate you approving this milestone💯, that will be all for now.

**Pull request merged**